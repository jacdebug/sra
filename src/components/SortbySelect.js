import React from 'react';
import styled from 'styled-components';

const SortbySelect = ({ sortBy, sortOnChange }) =>
  <Container>
    <label htmlFor="sort-select">Sort by: </label>
    <select id="sort-select" onChange={sortOnChange}>
      {[
        { id: 'id', label: 'Id' },
        { id: 'name', label: 'Name' },
        { id: 'size', label: 'Size' },
        { id: 'ext', label: 'Ext' },
        { id: 'created', label: 'Created' },
      ].map(ele => {
        return sortBy === ele.id
          ? <option selected value={ele.id}>
              {ele.label}
            </option>
          : <option value={ele.id}>
              {ele.label}
            </option>;
      })}
    </select>
  </Container>;

const Container = styled.div`margin: 4px 0 0;`;
export default SortbySelect;
