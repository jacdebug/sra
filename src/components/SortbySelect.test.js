import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SortbySelect from './SortbySelect';

it('renders without crashing', () => {
  const mockFn = jest.fn();
  shallow(<SortbySelect sortBy="id" sortOnChange={mockFn} />);
});

it('renders correctly', () => {
  const mockFn = jest.fn();
  const tree = shallow(<SortbySelect sortBy="id" sortOnChange={mockFn} />);
  expect(toJson(tree)).toMatchSnapshot();
});
