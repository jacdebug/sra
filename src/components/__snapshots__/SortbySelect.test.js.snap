// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <label
    htmlFor="sort-select"
  >
    Sort by: 
  </label>
  <select
    id="sort-select"
    onChange={[Function]}
    value="id"
  >
    <option
      value="id"
    >
      Id
    </option>
    <option
      value="name"
    >
      Name
    </option>
    <option
      value="size"
    >
      Size
    </option>
    <option
      value="ext"
    >
      Ext
    </option>
    <option
      value="created"
    >
      Created
    </option>
  </select>
</styled.div>
`;
